neuraxle/base.py

```diff
@@ -126,6 +129,13 @@ def __init__(
 
         self.pending_mutate: ('BaseStep', str, str) = (None, None, None)
         self.is_initialized = False
+        self.parent_step = None
+
+    def set_parent(self, parent_step: 'BaseStep'):
+        self.parent_step = parent_step
+
+    def save(self, data_container: DataContainer):
+        self.parent_step.save(data_container)
```

Comment:
Perhaps something like `self.saver.save(self, data_container)`?

This needs more thinking on the OOP design side of things. Could we come up with many different ideas to pick the best one before moving more forward?